src/status_im/extensions/ethereum.cljs

```diff
@@ -98,8 +98,119 @@
 
 (handlers/register-handler-fx
  :extensions/ethereum-call
- (fn [{db :db} [_ _ {:keys [to] :as arguments}]]
-   (wrap-with-resolution db arguments :to execute-ethcall)))
+ (fn [{db :db} [_ _ {:keys [address] :as arguments}]]
+   (wrap-with-resolution db arguments :address execute-ethcall)))
+
+(handlers/register-handler-fx
+ :extensions/ethereum-erc20-total-supply
+ (fn [{db :db} [_ _ {:keys [address] :as arguments}]]
+   (let [arguments' (merge arguments
+                           {:address address
+                            :method "totalSupply()"
+                            :outputs ["uint256"]})]
+     (wrap-with-resolution db arguments' :address execute-ethcall))))
+
+(handlers/register-handler-fx
+ :extensions/ethereum-erc20-balance-of
+ (fn [{db :db} [_ _ {:keys [address token-owner] :as arguments}]]
+   (let [arguments' (merge arguments
+                           {:address address
+                            :method "balanceOf(address)"
+                            :params [token-owner]
+                            :outputs ["uint256"]})]
+     (wrap-with-resolution db arguments' :address execute-ethcall))))
+
+(handlers/register-handler-fx
+ :extensions/ethereum-erc20-allowance
+ (fn [{db :db} [_ _ {:keys [address token-owner spender] :as arguments}]]
+   (let [arguments' (merge arguments
+                           {:address address
+                            :method "allowance(address,address)"
+                            :params [token-owner spender]
+                            :outputs ["uint256"]})]
+     (wrap-with-resolution db arguments' :address execute-ethcall))))
+
+(handlers/register-handler-fx
+ :extensions/ethereum-erc20-transfer
+ (fn [{db :db} [_ _ {:keys [address to value] :as arguments}]]
+   (let [arguments' (merge arguments
+                           {:to address
+                            :method "transfer(address,uint256)"
+                            :params [to value]
+                            :value nil})]
+     (wrap-with-resolution db arguments' :to execute-send-transaction))))
+
+(handlers/register-handler-fx
+ :extensions/ethereum-erc20-transfer-from
+ (fn [{db :db} [_ _ {:keys [address from to value] :as arguments}]]
+   (let [arguments' (merge arguments
+                           {:to address
+                            :method "transferFrom(address,address,uint256)"
+                            :params [from to value]
+                            :value nil})]
```

Comment:
Do we need those `value nil` ?

Reply:
yes, because `value` for `send-transaction` for an ETH transfer, here we use it as a parameter in a smartcontract method call. as said @flexsurfer to avoid confusion i make a new map here.

Reply:
Not following 😅  So just omitting value would not work?